src/plugins/data/public/search/search_interceptor.ts

```diff
@@ -162,30 +165,40 @@ export class SearchInterceptor {
     combinedSignal.addEventListener('abort', cleanup);
 
     return {
-      combinedSignal,
       timeoutSignal,
+      combinedSignal,
       cleanup,
     };
   }
 
+  private showTimeoutErrorToast = (e: SearchTimeoutError, sessionId?: string) => {
+    this.deps.toasts.addDanger({
+      title: 'Timed out',
+      text: toMountPoint(e.getErrorMessage(this.application)),
+    });
+  };
+
+  private showTimeoutErrorMemoized = memoize(
+    this.showTimeoutErrorToast,
+    (_: SearchTimeoutError, sessionId: string) => {
+      return sessionId;
+    }
+  );
+
   /**
-   * Right now we are throttling but we will hook this up with background sessions to show only one
-   * error notification per session.
+   * Show one error notification per session.
    * @internal
    */
-  private showTimeoutError = debounce(
-    (e: SearchTimeoutError) => {
-      this.deps.toasts.addDanger({
-        title: 'Timed out',
-        text: toMountPoint(e.getErrorMessage(this.application)),
-      });
-    },
-    30000,
-    { leading: true, trailing: false }
-  );
+  private showTimeoutError = (e: SearchTimeoutError, sessionId?: string) => {
+    if (sessionId) {
+      this.showTimeoutErrorMemoized(e, sessionId);
+    } else {
+      this.showTimeoutErrorToast(e, sessionId);
+    }
+  };
 
   /**
-   * Searches using the given `search` method. Overrides the `AbortSignal` with one that will abort
+   * Searches using theג given `search` method. Overrides the `AbortSignal` with one that will abort
```

Comment:
Somehow a `ג` got added in this comment 🤣

Reply:
That's a Hebrew `g`, mirroring the `g` in "given" :-D